include/alpaka/core/ApiHipRt.hpp

```diff
@@ -83,7 +83,7 @@ namespace alpaka
         static constexpr DeviceAttr_t deviceAttributeMaxThreadsPerBlock = ::hipDeviceAttributeMaxThreadsPerBlock;
         static constexpr DeviceAttr_t deviceAttributeMultiprocessorCount = ::hipDeviceAttributeMultiprocessorCount;
 
-#    if BOOST_LANG_HIP >= BOOST_VERSION_NUMBER(4, 5, 0)
+#    if HIP_VERSION >= 40500000
```

Comment:
Regarding to my last point, I am confused why you use `HIP_VERSION` and not `BOOST_LANG_VERSION` as before.
What was wrong with the original macro comparison?

Reply:
The problem is that `BOOST_LANG_HIP` is defined only when using `hipcc` to compile host or device code, but it does not get defined when using a different compiler (_e.g._ `gcc`) to compile HIP host code.

The same can be done using the Boost version macros:
```suggestion
#    if BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0) >= BOOST_VERSION_NUMBER(4, 5, 0)
```

I can change it to this approach if it is preferred (and then I guess I should make the same change in `include/alpaka/core/ApiCudaRt.hpp` with respect to `CUDART_VERSION`).

Reply:
Using `HIP_VERSION` is more compact :-)

Reply:
IMO we should keep `if BOOST_LANG_HIP >= BOOST_VERSION_NUMBER(4, 5, 0)` to have a single place in the code where the version number stored, in our case it is `#include <alpaka/core/BoostPredef.hpp>`

Reply:
We discussed this topic in the dev meeting and we should unify the creation of the version number in `BoostPredef.hpp` so that we do not have `BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0)` somewhere in the code expect in the predef header.

`BOOST_LANG_HIP` should be set always if `HIP_VERSION_MAJOR` is available.

Reply:
This is the only solution that _does not work_ .

`BOOST_LANG_HIP` is defined only when compiling with `hipcc`, not when compiling with `gcc`, so it is not a suitable macro to use here.

`BOOST_LANG_HIP`'s behaviour cannot be changed, as it is used elsewhere specifically to check for a compiler with HIP support.

The viable options are
  - use `HIP_VERSION >= 40500000`
  - use `BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0) >= BOOST_VERSION_NUMBER(4, 5, 0)`
  - `#define BOOST_HIP_VERSION = BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0)` and check `BOOST_HIP_VERSION >= BOOST_VERSION_NUMBER(4, 5, 0)`

Reply:
> `BOOST_LANG_HIP` should be set always if `HIP_VERSION_MAJOR` is available.

Maybe - but that is a _huge_ change in how Alpaka headers are written, and a departure from how Boost's own `BOOST_LANG_CUDA` behaves: `BOOST_LANG_CUDA` is defined only when compiling a `.cu` file with `nvcc` (or `clang` in cuda mode).

Reply:
OK, in that case, keep it how it is for now and we will postpone it to a future PR.